shark_turbine/kernel/wave/wave.py

```diff
@@ -209,11 +218,66 @@ def _trace_and_get_kernel_signature(
         if kwargs.get("canonicalize", False):
             canonicalize_module(mb.module_op)
 
-        return mb, graph
+        return mb, graph, exe, kernel_sig, entrypoint_name
 
     def test_execute(self, args, kwargs):
         # For now only tracing
-        mb, graph = self._trace_and_get_kernel_signature(args, kwargs)
+        (
+            mb,
+            graph,
+            exe,
+            kernel_sig,
+            entrypoint_name,
+        ) = self._trace_and_get_kernel_signature(args, kwargs)
+
+        if kwargs.get("run", False):
+            # TODO: cache compiled code
+            host_codegen.isolated_test_call(mb, exe, kernel_sig, entrypoint_name)
+            asm = mb.module_op.get_asm()
+
+            kernel_inputs = []
+            kernel_outputs = []
+            for arg, b in zip(args, kernel_sig.kernel_buffer_bindings):
+                usage = b.kernel_buffer_type.usage
+                if usage == kernel_codegen.KernelBufferUsage.INPUT:
+                    kernel_inputs.append(arg)
+
+                if usage == kernel_codegen.KernelBufferUsage.OUTPUT:
+                    kernel_outputs.append(arg)
+
+            # TODO: Unhardcode
+            target = "gfx942"
+            flags = [
+                "--iree-hal-target-backends=rocm",
+                f"--iree-rocm-target-chip={target}",
+                "--iree-vm-bytecode-module-strip-source-map=true",
+                "--iree-opt-strip-assertions=true",
+                "--iree-vm-target-truncate-unsupported-floats",
+            ]
+            if kwargs.get("print_ir_after_all", False):
+                flags.append("--mlir-print-ir-after-all")
+
+            res = compile_str(asm, target_backends=["rocm"], extra_args=flags)
+
+            config = rt.Config("hip")
+            mod = rt.VmModule.copy_buffer(config.vm_instance, res)
```

Comment:
Can we use a wrap_buffer here instead of copy buffer?

Reply:
`wrap_buffer` fails with `ValueError: VmModule.from_aligned_memory received an unaligned buffer. Got 0x0x55871ff26120, expected alignment 64`

Reply:
good to know, thanks for checking.